File: steel-agent-web/src/app/types/tender.ts
// ============================================================
// 招标信息相关 TypeScript 类型定义
// 对应后端 /api/v1/tenders/* 接口
// ============================================================

/** 招标详情 */
export interface TenderDetail {
  id: string;
  title: string;
  status: 'open' | 'closed' | 'won' | 'lost';
  budget: number;
  description: string;
  deadline: string;
  bid_deadline: string;
  source_url: string;
  region: string;
  category: string;
}

File: backend/internal/model/tender.go
package model

import "time"

// Tender represents a procurement tender record.
type Tender struct {
	ID          uint      `gorm:"column:id;primaryKey" json:"id"`
	Title       string    `gorm:"column:title" json:"title"`
	Region      string    `gorm:"column:region" json:"region"`
	Category    string    `gorm:"column:category" json:"category"`
	Budget      float64   `gorm:"column:budget" json:"budget"`
	Deadline    time.Time `gorm:"column:deadline" json:"deadline"`
	BidDeadline time.Time `gorm:"column:bid_deadline" json:"bid_deadline"`
	Status      string    `gorm:"column:status" json:"status"`
	SourceURL   string    `gorm:"column:source_url" json:"source_url"`
	Description string    `gorm:"column:description" json:"description"`
	CreatedAt   time.Time `gorm:"column:created_at" json:"created_at"`
}

// TableName returns the database table name for Tender.
func (Tender) TableName() string {
	return "tenders"
}
